dhall/src/Dhall/Pretty/Internal.hs

```diff
@@ -139,29 +139,28 @@ isWhitespace c =
 
     >>> let unusedSourcePos = Text.Megaparsec.SourcePos "" (Text.Megaparsec.mkPos 1) (Text.Megaparsec.mkPos 1)
     >>> let nonEmptySrc = Src unusedSourcePos unusedSourcePos "-- Documentation for x\n"
-    >>> "let" <> renderSrc (Just nonEmptySrc) " " <> "x = 1 in x"
+    >>> "let" <> " " <> renderSrc (Just nonEmptySrc) id <> "x = 1 in x"
     let -- Documentation for x
         x = 1 in x
     >>> let emptySrc = Src unusedSourcePos unusedSourcePos "      "
-    >>> "let" <> renderSrc (Just emptySrc) " " <> "x = 1 in x"
+    >>> "let" <> " " <> renderSrc (Just emptySrc) id <> "x = 1 in x"
     let x = 1 in x
-    >>> "let" <> renderSrc Nothing " " <> "x = 1 in x"
+    >>> "let" <> " " <> renderSrc Nothing id <> "x = 1 in x"
     let x = 1 in x
 -}
 renderSrc
     :: Maybe Src
     -- ^ Source span to render (if present)
+    -> (Text -> Text)
+    -- ^ Used to preprocess the comment string (e.g. to strip whitespace)
```

Comment:
It seems to me that it would be more "conventional" to have the function as the first parameter, and the `Maybe Src` as the second, but I'm not sure whether it's a good idea in this case.

Reply:
Hmm, now that you mention it, the ordering is a bit ad-hoc.